src/main/java/com/checkmarx/flow/service/JiraService.java

```diff
@@ -957,91 +954,121 @@ private String getBody(ScanResults.XIssue issue, ScanRequest request, String fil
             }
         }
+    }
 
-        return ImmutableMap.of(
-                "new", newIssues,
-                "updated", updatedIssues,
-                "closed", closedIssues
-        );
+    private void createIssue(ScanRequest request, List<String> newIssues, Map.Entry<String, ScanResults.XIssue> xIssue, ScanResults.XIssue currentIssue) throws JiraClientException {
+        log.debug("Creating new issue with key {}", xIssue.getKey());
+        String newIssue = this.createIssue(currentIssue, request);
+        newIssues.add(newIssue);
+        log.info("New issue created. #{}", newIssue);
+    }
+
+    private void updateIssue(ScanRequest request, List<String> updatedIssues, Map.Entry<String, ScanResults.XIssue> xIssue, ScanResults.XIssue currentIssue, Issue issue) throws JiraClientException {
+        log.debug("Issue still exists.  Updating issue with key {}", xIssue.getKey());
+        Issue updatedIssue = this.updateIssue(issue.getKey(), currentIssue, request);
+        if (updatedIssue != null) {
+            log.debug("Update completed for issue #{}", updatedIssue.getKey());
+            updatedIssues.add(updatedIssue.getKey());
+            if (jiraProperties.isUpdateComment() && !ScanUtils.empty(jiraProperties.getUpdateCommentValue())) {
+                addCommentToBug(issue.getKey(), jiraProperties.getUpdateCommentValue());
+            }
+        }
+    }
+
+    private void closeIssueInCaseOfIssueIsInOpenState(ScanRequest request, List<String> closedIssues, Issue fpIssue) throws JiraClientException {
+        if (request.getBugTracker().getOpenStatus().contains(fpIssue.getStatus().getName())) { //If the status is of open state, close it
+            /*Close the issue*/
+            log.info("Closing issue with key {}", fpIssue.getKey());
+            this.transitionCloseIssue(fpIssue.getKey(), request.getBugTracker().getCloseTransition(), request.getBugTracker(), true);
+            closedIssues.add(fpIssue.getKey());
+        }
+    }
+
+    private Issue checkForFalsePositiveIssuesInList(ScanRequest request, Map.Entry<String, ScanResults.XIssue> xIssue, ScanResults.XIssue currentIssue, Issue issue) throws JiraClientException {
+        Issue fpIssue;
+        if (flowProperties.isListFalsePositives()) { //Update the ticket if flag is set
+            log.debug("Issue is being updated to reflect false positive references.  Updating issue with key {}", xIssue.getKey());
+            fpIssue = this.updateIssue(issue.getKey(), currentIssue, request);
+        } else { //otherwise simply get a reference to the issue
+            fpIssue = this.getIssue(issue.getKey());
+        }
+        return fpIssue;
+    }
+
+    private void getAndModifiedRequestApplication(ScanRequest request) {
+        String application = request.getApplication();
+        if (!ScanUtils.empty(application)) {
+            application = application.replaceAll("[^a-zA-Z0-9-_.+]+", "_");
```

Comment:
"[^a-zA-Z0-9-_.+]+" and "_" should be defined as const strings in the Class
I would also add a short comment of example explaining what is replaced with what.